src/modules/launcher/Wox.Infrastructure/Storage/JsonStorage`1.cs

```diff
@@ -20,11 +22,13 @@ public class JsonStorage<T>
         private static readonly IPath Path = FileSystem.Path;
         private static readonly IFile File = FileSystem.File;
 
+        private readonly object _saveLock = new object();
+
         // use property initialization instead of DefaultValueAttribute
         // easier and flexible for default value of object
         private static readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions
         {
-            DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull,
+            DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.Never,
```

Comment:
Is there an ither way to achieve the same result? Changing this might waste space for "null" value data on the disk.

Reply:
When we do not write the null values ​​to the information file, when we compare the file related to the information file, we may see incompatibility and delete it even if there is no change in the version updates. That's why I changed DefaultIgnoreCondition that way. In addition, Null fields are less in number than filled ones and do not take up a large area. Also, as a user, I discovered areas that I had not noticed before. I think it can also provide useful information to those who examine these files in terms of usage.

Reply:
Regarding the Plugin options it make a difference of half of the used space and it makes the file better to read/view.

Please validate that the plugin options still don't write Null values to json.

Reply:
The behavior of JSON files has been changed back. A new JsonSerializerOptions has been created for information files. All JSON fields are kept and checked only in information files.